src/mraa.c

```diff
@@ -1130,6 +1140,32 @@ mraa_link_targets(const char* filename, const char* targetname)
     }
 }
 
+mraa_result_t
+mraa_find_uart_bus_pci(const char* pci_dev_path, char** dev_name)
+{
+    char path[PATH_MAX];
+    int max_allowable_len = 16;
+    snprintf(path, PATH_MAX-1, "%s", pci_dev_path);
+    if (!mraa_file_exist(path)) return MRAA_ERROR_UNSPECIFIED;
+
+    struct dirent **namelist;
+    int n = scandir(path, &namelist, NULL, alphasort);
+    if (n <= 0) {
+        syslog(LOG_ERR, "Failed to find expected UART bus");
+        return MRAA_ERROR_NO_RESOURCES;
+    }
+
+    *dev_name = (char *) malloc(sizeof(char)*max_allowable_len);
+
+    snprintf(*dev_name, max_allowable_len,"/dev/%s" ,namelist[n-1]->d_name);
+    while (n--) {
+        free(namelist[n]);
+    }
+    free(namelist);
+    syslog(LOG_INFO, "UART device: %s selected for Initialization", *dev_name);
```

Comment:
nitpick :smiley: : no need to capitalize "initialization"

Reply:
done.